spec/services/store_sp_metadata_in_session_spec.rb

```diff
@@ -2,143 +2,392 @@
-          phishing_resistant_requested: true,
-          ial: 2,
-          ial2: true,
-          ialmax: false,
-          request_url: 'http://issuer.gov',
-          request_id: request_id,
-          requested_attributes: %w[email],
-          biometric_comparison_required: false,
-        }
-
-        instance.call
-        expect(app_session[:sp]).to eq app_session_hash
+          it 'sets the session[:sp] hash correctly' do
+            expect(app_session[:sp]).to eq(
+              {
+                issuer: issuer,
+                aal_level_requested: 3,
+                piv_cac_requested: false,
+                phishing_resistant_requested: true,
+                ial: 2,
+                ial2: true,
+                ialmax: false,
+                request_url: request_url,
+                request_id: request_id,
+                requested_attributes: requested_attributes,
+                biometric_comparison_required: true,
+              },
+            )
+          end
+        end
       end
-    end
 
-    context 'when biometric comparison is requested' do
-      it 'sets the session[:sp] hash' do
-        app_session = {}
-        request_id = SecureRandom.uuid
-        ServiceProviderRequestProxy.find_or_create_by(uuid: request_id) do |sp_request|
-          sp_request.issuer = 'issuer'
-          sp_request.ial = Saml::Idp::Constants::IAL2_AUTHN_CONTEXT_CLASSREF
-          sp_request.aal = Saml::Idp::Constants::AAL3_AUTHN_CONTEXT_CLASSREF
-          sp_request.url = 'http://issuer.gov'
-          sp_request.requested_attributes = %w[email]
-          sp_request.biometric_comparison_required = true
-        end
-        instance = StoreSpMetadataInSession.new(session: app_session, request_id: request_id)
-
-        app_session_hash = {
-          issuer: 'issuer',
-          aal_level_requested: 3,
-          piv_cac_requested: false,
-          phishing_resistant_requested: true,
-          ial: 2,
-          ial2: true,
-          ialmax: false,
-          request_url: 'http://issuer.gov',
-          request_id: request_id,
-          requested_attributes: %w[email],
-          biometric_comparison_required: true,
-        }
-
-        instance.call
-        expect(app_session[:sp]).to eq app_session_hash
+      # new-style SP requests
+      context 'and the `use_vot_in_sp_requests` config flag is true' do
+        let(:issuer) { 'issuer' }
+        let(:request_url) { 'http://issuer.gov' }
+        let(:requested_attributes) { %w[email] }
+        let(:request_acr) { nil }
+        let(:request_vtr) { [] }
+
+        before do
+          allow(IdentityConfig.store).to receive(:use_vot_in_sp_requests).and_return(true)
+
+          sp_request = ServiceProviderRequestProxy.find_or_create_by(uuid: request_id) do |sp_request|
+            sp_request.issuer = issuer
+            sp_request.url = request_url
+            sp_request.requested_attributes = requested_attributes
+            sp_request.acr_values = request_acr
```

Comment:
It looks like `request_acr` never gets set anywhere. It should be the `ial` and `aal` strings separated by a space (` `) when ACR values are used.

Reply:
Added specs for this.